### PR TITLE
17580 500 when posting to textile with json

### PR DESCRIPTION
**Pull request by hschink**
Implements the following tickets (the commits also reflect this order):
1. https://community.openproject.org/work_packages/17580
2. https://community.openproject.org/work_packages/17439
3. https://community.openproject.org/work_packages/17440

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://fix-17580-500-when-posting-to-textile-with-json.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: lib/api/v3/render/render_api.rb [hunk 1]
```diff
@@ -37,6 +37,22 @@ class RenderAPI < Grape::API
         resources :render do
           helpers do
             SUPPORTED_CONTEXT_NAMESPACES = ['work_packages'].freeze
+            SUPPORTED_MEDIA_TYPE = 'text/plain'
+
+            def check_content_type
```
Comment (myabc) [inline lib/api/v3/render/render_api.rb:42]: @hschink I'm surprised Grape doesn't have a mechanism for this built in. Can you not declare an additional `content_type :txt, 'text/plain'` just for this resource?

Comment (hschink) [inline lib/api/v3/render/render_api.rb:42]: @myabc With `content_type :txt, 'text/plain'`set the API still accepts different content types. For me it looks like Grape hasn't implemented a content type check.

Comment (myabc) [inline lib/api/v3/render/render_api.rb:42]: @hschink is this part of the Grape `README` relevant?

> When the content-type is omitted, Grape will return a 406 error code unless `default_format` is specified.
> When the content-type is omitted, Grape will return a 406 error code unless `default_format` is specified.
> The following API will try to parse any data without a content-type using a JSON parser.
> 
> ``` ruby
> class Twitter::API < Grape::API
>   format :json
>   default_format :json
> end
> ```

Comment (myabc) [inline lib/api/v3/render/render_api.rb:42]: @hschink looks like you're right. This PR might help us though: https://github.com/intridea/grape/pull/806

**Pull request merged**